src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/MultiDimensionalExtension.java

```diff
@@ -288,6 +417,47 @@ private int getExpandlimit(
         }
     }
 
+    private int getLimit(ResourceInfo resource, Object clientLimit, HttpServletResponse response)
+            throws OWSException {
+        if (clientLimit != null) {
+            Integer value = Converters.convert(clientLimit, Integer.class);
+            if (value == null) {
+                throw new OWSException(
+                        400,
+                        "InvalidParameterValue",
+                        LIMIT,
+                        "Invalid "
+                                + LIMIT
+                                + " "
+                                + "value "
+                                + clientLimit
+                                + ", expected an integer value");
+            }
+            int limitMax = LIMIT_MAX_DEFAULT;
+            if (value < limitMax) {
+                return value;
+            } else {
+                String message =
+                        "Client provided a limit value higher than the configured max, using the "
+                                + "internal value (provided/max): "
+                                + value
+                                + "/"
+                                + limitMax;
+                LOGGER.log(Level.INFO, message);
```

Comment:
Will it make sense to log it as an warning ?

Reply:
It would make it pop up in PRODUCTION Logs, which is not what we want, this is more of a debugging facility. Also, the other limit is handled the same way (INFO when client goes above)